src/test/java/org/folio/marc4ld/mapper/field776/Marc2Bibframe776IT.java

```diff
@@ -0,0 +1,259 @@
+package org.folio.marc4ld.mapper.field776;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.folio.ld.dictionary.PredicateDictionary.CREATOR;
+import static org.folio.ld.dictionary.PredicateDictionary.EXTENT;
+import static org.folio.ld.dictionary.PredicateDictionary.MAP;
+import static org.folio.ld.dictionary.PredicateDictionary.OTHER_VERSION;
+import static org.folio.ld.dictionary.PredicateDictionary.TITLE;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.AGENT;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.IDENTIFIER;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.ID_CODEN;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.ID_ISBN;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.ID_ISSN;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.ID_LCCN;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.ID_LOCAL;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.ID_STRN;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.ID_UNKNOWN;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.INSTANCE;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.WORK;
+import static org.folio.marc4ld.mapper.test.TestUtil.loadResourceAsString;
+import static org.folio.marc4ld.mapper.test.TestUtil.validateEdge;
+import static org.folio.marc4ld.mapper.test.TestUtil.validateResource;
+
+import java.util.List;
+import java.util.Map;
+import org.folio.ld.dictionary.ResourceTypeDictionary;
+import org.folio.marc4ld.Marc2LdTestBase;
+import org.folio.marc4ld.mapper.test.SpringTestConfig;
+import org.folio.marc4ld.service.marc2ld.bib.MarcBib2LdMapperImpl;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@EnableConfigurationProperties
+@SpringBootTest(classes = SpringTestConfig.class)
+class Marc2Bibframe776IT extends Marc2LdTestBase {
+
+  @Autowired
+  private MarcBib2LdMapperImpl marc2BibframeMapper;
+
+  @Test
+  void shouldMapField776Correctly() {
+    //given
+    var marc = loadResourceAsString("fields/776/marc_776.jsonl");
+
+    //when
+    var result = marc2BibframeMapper.fromMarcJson(marc);
+
+    //then
+    validateAllIds(result);
```

Comment:
Minor

this assertion create, but to complicated insupportion
for finding parts of expected values
I suggest separating by methods
-validateAgent
-validateIdentifier
-validateTitle

in this case instead of using an iterator, start checking real existing of edges types

example of what I mean
org.folio.marc4ld.authority.field100.MarcToBibframeAuthorityConcept100IT

Reply:
Done